pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -588,28 +590,37 @@ class Timestamp(_Timestamp):
             elif tz is not None:
                 raise ValueError('Can provide at most one of tz, tzinfo')
 
-        if ts_input is _no_input:
+        if is_string_object(ts_input):
+            # User passed a date string to parse.
+            # Check that the user didn't also pass a date attribute kwarg.
+            date_attrs = [year, month, day, hour, minute, second, microsecond,
```

Comment:
make this a class attr, ``_date_attributes``

Reply:
worth looking into: can class attributes be made readonly?  wouldn't want an unwitting user to change this.